app/controllers/projects_controller.rb

```diff
@@ -1,7 +1,16 @@
 class ProjectsController < ApplicationController
 
   def index
-    @projects = Project.where(active: true)
+    @projects = Project.where(active: true).order(:date)
+    if params[:sort_by] == 'a-z'
+      @projects = Project.where(active: true).order(:title)
+    elsif params[:sort_by] == 'needs_driver'
+      needs_driver = @projects.find_all do |project|
+        project.carpools.empty?
+      end
+      @projects = needs_driver 
+    else
+    end
```

Comment:
We could probably move this a class method on the Project model, which would take `params[:sort_by]` as an argument and return the sorted projects.

That way, we could move the conditional logic out of the Controller and into the Model.